### PR TITLE
[Search] Fix "SearchOptions.SemanticSearch.MaxWait" Issue

### PR DESCRIPTION
**Pull request by ShivangiReja**
There is an issue in the Search SDK, specifically in [here](https://github.com/Azure/azure-sdk-for-net/blob/feature/ShivangiReja/search2023-11-01/sdk/search/Azure.Search.Documents/src/Options/SearchOptions.cs#L226). Instead of utilizing `SemanticSearch?.MaxWait?.Milliseconds`, it should be replaced with `(int)SemanticSearch?.MaxWait?.TotalMilliseconds`.

When a user sets MaxWait to `TimeSpan.FromMilliseconds(1000)`, `SemanticSearch?.MaxWait?.Milliseconds` returns 0, while `(int)SemanticSearch?.MaxWait?.TotalMilliseconds` returns 1000.

The reason behind this change is that the TimeSpan format {00:00:01} does not represent milliseconds. This format conveys a TimeSpan in HH:mm:ss format.

In summary:
- **Milliseconds** returns the milliseconds component within a TimeSpan.
- **TotalMilliseconds** returns the total number of milliseconds represented by the TimeSpan.

### CODE REVIEW

File: sdk/search/Azure.Search.Documents/src/Options/SearchOptions.cs [hunk 1]
```diff
@@ -223,10 +223,17 @@ private string QueryCaptionRaw
...
         {
-            get { return SemanticSearch?.MaxWait?.Milliseconds; }
+            get
+            {
+                if (SemanticSearch?.MaxWait != null)
+                {
+                    return (int)SemanticSearch.MaxWait?.TotalMilliseconds;
+                }
+                return null;
+            }
```
Comment (heaths) [inline sdk/search/Azure.Search.Documents/src/Options/SearchOptions.cs:233]: Tip: you can simply this with a cast to `int?`: https://sharplab.io/#v2:C4LglgNgNAJiDUAfAAgJgIwFgBQOAqYAtgKYDKADgIYB2A/AATADO9AvPdQK4QQDcOYasAZg29ABSDhASma0AdHgD2wShACykCGCbEAxkuowmvIA

File: sdk/search/Azure.Search.Documents/tests/DocumentOperations/VectorSearchTests.cs [hunk 2]
```diff
@@ -150,6 +152,33 @@ await foreach (SearchResult<Hotel> result in response.GetResultsAsync())
...
+                            MaxWait = TimeSpan.FromMilliseconds(700),
+                        },
+                        QueryType = SearchQueryType.Semantic,
+                    }));
+
+            Assert.AreEqual("InvalidRequestParameter", ex.ErrorCode);
+            StringAssert.StartsWith(
+                "The 'semanticMaxWaitInMilliseconds' parameter needs to be larger than 750.",
+                ex.Message);
+        }
```
Comment (heaths) [inline sdk/search/Azure.Search.Documents/tests/DocumentOperations/VectorSearchTests.cs:180]: Why are we asserting service-specific behavior that can and will likely change? This seems to go against our [guidelines](https://azure.github.io/azure-sdk/general_implementation.html#general-params-server-validation).

**Review by heaths — APPROVED** [review]
